collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -225,6 +227,19 @@ protected void setupOSMDroid() {
         org.osmdroid.config.Configuration.getInstance().setUserAgentValue(userAgentProvider.getUserAgent());
     }
 
+    private void setupStrictMode() {
+        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
+                .detectAll()
+                .penaltyLog()
```

Comment:
I'm a little worried that this will get lost as Logcat is really noisy. Could we maybe use `penaltyDialog`? Given it's only on for debug builds I think it's fine for it to be loud!

Reply:
I tried that, but there were too many dialogs getting generated as soon as the app opens and causes the app to crash. Maybe we should try that once the known leaks are fixed. Thoughts?

Reply:
Could we fix those problems as part of this PR? I think if we're going to put a check in we should fix any existing problems the check would find.

Reply:
👍

Reply:
Thanks for picking this back up, @shobhitagarwal1612!

Agreed that would be ideal. That said, if it requires really invasive refactoring it may not be appropriate. Could you start by enumerating the issues you've identified and maybe a quick estimate of whether you think it's a straightforward fix?

Reply:
I also suspect that https://github.com/opendatakit/collect/issues/3705 would fix most of the issues occurring on load. Please see what you think, @shobhitagarwal1612, so we can avoid duplicate/conflicting changes.

Reply:
Looks like there's a leak in `osmdroid` library itself which we can't do anything about. 

Secondly, a lot disk read operations (shared prefs) happen on UI thread. That can be solved by moving all those calls to a non-UI thread. This looks like a big refactor to me

Reply:
Hmmm interesting that it picks up the shared prefs stuff. I'd say as long as we're calling `apply` instead of `commit` I'm not really worried about it for now (did some reason [here](https://stackoverflow.com/questions/4371273/should-accessing-sharedpreferences-be-done-off-the-ui-thread)). I think we can probably disable that check if possible?

Reply:
>  I think we can probably disable that check if possible?

Done